mediapipe/tasks/c/components/containers/embedding_result_converter.cc

```diff
@@ -0,0 +1,79 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "mediapipe/tasks/c/components/containers/embedding_result_converter.h"
+
+#include <cstdint>
+#include <cstdlib>
+
+#include "mediapipe/tasks/c/components/containers/embedding_result.h"
+#include "mediapipe/tasks/cc/components/containers/embedding_result.h"
+
+namespace mediapipe::tasks::c::components::containers {
+
+void CppConvertToEmbeddingResult(
+    const mediapipe::tasks::components::containers::EmbeddingResult& in,
+    EmbeddingResult* out) {
+  out->has_timestamp_ms = in.timestamp_ms.has_value();
+  out->timestamp_ms = out->has_timestamp_ms ? in.timestamp_ms.value() : 0;
+
+  out->embeddings_count = in.embeddings.size();
+  out->embeddings =
+      out->embeddings_count ? new Embedding[out->embeddings_count] : nullptr;
+
+  for (uint32_t i = 0; i < out->embeddings_count; ++i) {
+    auto embedding_in = in.embeddings[i];
+    auto& embedding_out = out->embeddings[i];
+
+    if (!embedding_in.float_embedding.empty()) {
+      embedding_out.values_count = embedding_in.float_embedding.size();
+      embedding_out.float_embedding =
+          embedding_out.values_count ? new float[embedding_out.values_count]
+                                     : nullptr;
+      std::copy(embedding_in.float_embedding.begin(),
+                embedding_in.float_embedding.end(),
+                embedding_out.float_embedding);
+      embedding_out.quantized_embedding = nullptr;
+    } else if (!embedding_in.quantized_embedding.empty()) {
+      embedding_out.values_count = embedding_in.quantized_embedding.size();
+      embedding_out.quantized_embedding =
+          embedding_out.values_count ? new char[embedding_out.values_count]
+                                     : nullptr;
+      std::copy(embedding_in.quantized_embedding.begin(),
+                embedding_in.quantized_embedding.end(),
+                embedding_out.quantized_embedding);
+      embedding_out.float_embedding = nullptr;
+    }
+
+    embedding_out.head_index = embedding_in.head_index;
+    embedding_out.head_name = embedding_in.head_name.has_value()
+                                  ? strdup(embedding_in.head_name->c_str())
+                                  : nullptr;
+  }
+}
+
+void CppCloseEmbeddingResult(EmbeddingResult* in) {
+  for (uint32_t i = 0; i < in->embeddings_count; ++i) {
+    auto embedding_in = in->embeddings[i];
+
+    delete[] embedding_in.float_embedding;
```

Comment:
Can you set the values to `nullptr`? I just merged this with https://github.com/google/mediapipe/commit/a72839ef990205898ea613e261d64525b9d25f79